src/components/forms/shipment/ShipmentOrderData.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useCallback, useState } from 'react';
+
+import { FieldValues } from 'react-hook-form';
+import { useIntl } from 'react-intl';
+
+import TableWithoutPagination from 'components/shared/table/TableWithoutPagination';
+import { ColumnType } from 'components/shared/table/types';
+import SubmitButton from 'shared/buttons/SubmitButton';
+import Dialog from 'shared/dialog/Dialog';
+
+import ShipmentOrderDetailsForm from './ShipmentOrderDetailsForm';
+
+function ShipmentOrderData({
+  onOrderDetailAdd, onOrderDetailDelete, orderDetails, mode,
+}: any) {
+  const { formatMessage } = useIntl();
+  const format = useCallback((id: string, values: any = '') => formatMessage({ id }, values), [formatMessage]);
+
+  const [displayAddModal, setDisplayAddModal] = useState(false);
+  const [displayShowOrderModal, setDisplayShowOrderModal] = useState(false);
+  const [orderDetailList, setOrderDetailList] = useState(orderDetails);
+  const [orderLineItemToShow, setOrderLineItemToShow] = useState([]);
+
+  const toggleAddModal = () => {
+    setDisplayAddModal(!displayAddModal);
+  };
+
+  const toggleShowOrderModal = (object?: FieldValues) => {
+    if (object) {
+      const lineItems = orderDetails.filter(
+        (order: any) => order.order_details_id === object.order_details_id,
+      ).map((order: any) => order.line_items);
+      setOrderLineItemToShow(lineItems);
+    }
+    setDisplayShowOrderModal(!displayShowOrderModal);
+  };
+
+  const onSubmitAdd = (formValues: FieldValues) => {
+    onOrderDetailAdd(formValues);
+    setOrderDetailList(orderDetails);
+    toggleAddModal();
+  };
+
+  const onDelete = (object: FieldValues) => {
+    onOrderDetailDelete(object);
+    const arrayDeleted = orderDetails.filter(
+      (order: any) => order.order_details_id !== object.order_details_id,
+    );
+    setOrderDetailList(arrayDeleted);
+  };
+
+  const columns: ColumnType[] = React.useMemo(() => [
+    {
+      Header: format('shipment.customer_name.label'),
+      accessor: 'customer_name',
+    },
+    {
+      Header: format('shipment.order_number.label'),
+      accessor: 'order_details_id',
+    },
+  ], [format]);
+
+  const editInitialValue = () => orderDetails.map(
+    (object: any) => ({ id: object.customer, name: object.customer_name }),
+  )[0];
+
+  return (
+    <>
+      <div className="flow-root">
+        <p className="float-left text-[21px] text-transit-black font-semibold">{format('shipment.orders.label')}</p>
+        <SubmitButton onClick={toggleAddModal} className="float-right h-fit w-fit" title={format('shipment.order_add.label')} />
+      </div>
```

Comment:
If you want to have title and button an different side you just need to add `flex flex-row justify-between` class to the `div` above and then css class are unnecessary in `p` and `SubmitButton` - it's easier